src/systems/host.cairo

```diff
@@ -84,12 +82,36 @@ mod host {
             game_id
         }
 
-        fn join(
-            self: @ContractState, world: IWorldDispatcher, game_id: u32, player_name: felt252,
+        fn set_max_players(
+            self: @ContractState, world: IWorldDispatcher, game_id: u32, player_count: u8
         ) {
             // [Setup] Datastore
             let mut store: Store = StoreTrait::new(world);
 
+            // [Check] Caller is the host
+            let mut game = store.game(game_id);
+            let caller = get_caller_address();
+            assert(caller == game.host, errors::HOST_CALLER_IS_NOT_THE_HOST);
+
+            assert(player_count >= MINIMUM_MAX_PLAYERS, errors::HOST_MAX_NB_PLAYERS_IS_TOO_LOW);
+
+            let mut addresses = array![];
+            let mut players = store.players(game);
+            loop {
```

Comment:
This loop is done 3 times at the end:
- In the store
- In this function
- In the`Game::set_max_players` function

This is very inefficient and players will pay this in fee.
We can either:
- Remove the `set_max_players` and consider only having a hard coded max player with the ability to start an unfilled game
- Skip this loop and just provide a Span<Player> to `Game::set_max_players`